query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -173,4 +254,17 @@ public ResultSetMetadata getResultSetMetadata() {
   public LinkedHashSet<Expression> getAllSelections() {
     return this.allSelections;
   }
+
+  public Optional<Double> getTimeSeriesColumn(String columnName) {
+    if (this.timeSeriesColumnMap.containsKey(columnName)) {
+      return Optional.of(Double.parseDouble(this.timeSeriesColumnMap.get(columnName)));
+    }
+    return Optional.empty();
+  }
+
+  public double getTimeFilterDuration(String columnName) {
+    long durationInMillis =
+        this.timeFilterMap.get(columnName).get(1) - this.timeFilterMap.get(columnName).get(0);
```

Comment:
Need to use Math.abs as it is not guaranteed while parsing that index `1` will have a larger value.